client/src/pages/Itinerary.js

```diff
@@ -1,5 +1,115 @@
+import { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import styled from 'styled-components/macro';
 import { Headline } from './Sights';
 
-export default function Itinerary() {
-  return <Headline>My Itinerary</Headline>;
+import deleteIcon from '../images/delete.svg';
+import infoIcon from '../images/info.svg';
+
+export default function Itinerary({ dateSightCombos, setDateSightCombos }) {
+  const [dates, setDates] = useState([]);
+
+  useEffect(() => {
+    function getDatesForItinerary() {
+      const allDates = dateSightCombos.map((combo) => combo.date);
+      let uniqueDates = [];
+      const map = {};
+      allDates.forEach((date) => {
+        if (!map[JSON.stringify(date)]) {
+          map[JSON.stringify(date)] = true;
+          uniqueDates.push(date);
+          uniqueDates.sort((a, b) => {
+            a = a.split('/');
+            b = b.split('/');
+            return a[2] - b[2] || a[1] - b[1] || a[0] - b[0];
+          });
+        }
+      });
+      return uniqueDates;
+    }
+    setDates(getDatesForItinerary());
+  }, [dateSightCombos]);
+
+  function removeFromItinerary(comboToRemove) {
+    const remaining = dateSightCombos.filter(
+      (combo) =>
+        (combo.sight._id && combo.date) !==
+        (comboToRemove.sight._id && comboToRemove.date)
+    );
+    setDateSightCombos(remaining);
+  }
```

Comment:
When I have more than one sight on a single day in my calendar and want to remove one of them, it deletes all of them. I would assume it has to do with this function?